<a href="https://colab.research.google.com/github/johacagua-mina/JohannaNova/blob/main/Task2_MSO4130.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# MSO4130 Portfolio Task 2
**Student:** Johanna Cagua Mina  
**Student ID:** M01089122

## Objective
This notebook answers Portfolio Task 2 using Python.

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats
import statsmodels.formula.api as smf

In [2]:
from google.colab import files
uploaded = files.upload()

Saving LWDATA_M01089122.csv to LWDATA_M01089122.csv


In [3]:
df = pd.read_csv("LWDATA_M01089122.csv")
df.head()

,SIZE,TURNOVER,LONDON,LW,ECEMPLOYEES,ABSENT
0,157.0,1239694.0,0.0,0.0,82.0,5.0
1,443.0,3348257.0,1.0,0.0,258.0,9.4
2,620.0,4940407.0,0.0,0.0,407.0,5.7
3,305.0,513637.0,0.0,0.0,173.0,5.6
4,173.0,595205.0,0.0,0.0,104.0,4.6


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   SIZE         100 non-null    float64
 1   TURNOVER     100 non-null    float64
 2   LONDON       100 non-null    float64
 3   LW           100 non-null    float64
 4   ECEMPLOYEES  100 non-null    float64
 5   ABSENT       100 non-null    float64
dtypes: float64(6)
memory usage: 4.8 KB


In [5]:
df["PECEMPLOYEES"] = (df["ECEMPLOYEES"] / df["SIZE"]) * 100
df.head()

,SIZE,TURNOVER,LONDON,LW,ECEMPLOYEES,ABSENT,PECEMPLOYEES
0,157.0,1239694.0,0.0,0.0,82.0,5.0,52.229299
1,443.0,3348257.0,1.0,0.0,258.0,9.4,58.239278
2,620.0,4940407.0,0.0,0.0,407.0,5.7,65.645161
3,305.0,513637.0,0.0,0.0,173.0,5.6,56.721311
4,173.0,595205.0,0.0,0.0,104.0,4.6,60.115607


## Part (a)

The dataset was successfully imported into Python using pandas.

A new variable, `PECEMPLOYEES`, was created by dividing the number of early-career employees (`ECEMPLOYEES`) by the total company size (`SIZE`) and multiplying by 100.

This represents the percentage of early-career employees in each company.

In [6]:
cont_table = pd.crosstab(df["LONDON"], df["LW"])
cont_table

LW,0.0,1.0
LONDON,,
0.0,24,28
1.0,22,26


In [7]:
chi2, p, dof, expected = stats.chi2_contingency(cont_table)

print("Chi-square:", chi2)
print("p-value:", p)
print("Degrees of freedom:", dof)
print("Expected frequencies:")
print(expected)

Chi-square: 0.0
p-value: 1.0
Degrees of freedom: 1
Expected frequencies:
[[23.92 28.08]
 [22.08 25.92]]


## Part (b) Interpretation

The chi-square test produced a p-value of 1.0, which is significantly greater than the 0.05 significance level.

Therefore, we fail to reject the null hypothesis.

This indicates that there is no evidence of an association between `LONDON` and `LW` in this dataset.

In other words, whether a company is located in London appears to be independent of whether it pays the Living Wage.

In [8]:
model1 = smf.ols("ABSENT ~ SIZE + TURNOVER + LW", data=df).fit()
print(model1.summary())

                            OLS Regression Results                            
Dep. Variable:                 ABSENT   R-squared:                       0.160
Model:                            OLS   Adj. R-squared:                  0.134
Method:                 Least Squares   F-statistic:                     6.085
Date:                Tue, 17 Mar 2026   Prob (F-statistic):           0.000778
Time:                        14:09:50   Log-Likelihood:                -222.56
No. Observations:                 100   AIC:                             453.1
Df Residuals:                      96   BIC:                             463.5
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      7.0962      0.483     14.681      0.0

## Part (c) Regression Interpretation

The regression results show that `LW` (Living Wage) has a statistically significant negative effect on absenteeism.

The coefficient for `LW` is approximately -1.926, with a p-value of 0.000, indicating strong statistical significance.

This suggests that, holding SIZE and TURNOVER constant, firms that pay the Living Wage have, on average, around 1.93 fewer units of absenteeism compared to firms that do not.

In contrast, both SIZE and TURNOVER are not statistically significant, as their p-values are well above 0.05. This indicates that they do not have a meaningful effect on absenteeism in this model.

Overall, the results suggest that paying the Living Wage is associated with lower levels of absenteeism.

The R-squared value of 0.16 suggests that the model explains a relatively small proportion of the variation in absenteeism, indicating that other factors not included in the model may also play an important role.

Cond. No. = 8.66e+06

In [9]:
model2 = smf.ols("PECEMPLOYEES ~ SIZE + TURNOVER + LW", data=df).fit()
print(model2.summary())

                            OLS Regression Results                            
Dep. Variable:           PECEMPLOYEES   R-squared:                       0.541
Model:                            OLS   Adj. R-squared:                  0.526
Method:                 Least Squares   F-statistic:                     37.67
Date:                Tue, 17 Mar 2026   Prob (F-statistic):           3.54e-16
Time:                        14:15:06   Log-Likelihood:                -327.96
No. Observations:                 100   AIC:                             663.9
Df Residuals:                      96   BIC:                             674.3
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     59.6759      1.387     43.033      0.0

## Part (d) Regression Interpretation

The regression results show that `LW` (Living Wage) has a statistically significant negative effect on the percentage of early-career employees (PECEMPLOYEES).

The coefficient for `LW` is approximately -13.58, with a p-value of 0.000, indicating strong statistical significance.

This suggests that, holding SIZE and TURNOVER constant, firms that pay the Living Wage have, on average, around 13.6 percentage points fewer early-career employees compared to firms that do not.

This implies that firms paying the Living Wage tend to retain more experienced employees, supporting the claim that Living Wage employers have lower staff turnover.

In contrast, SIZE and TURNOVER are not statistically significant, indicating that they do not have a meaningful effect on the proportion of early-career employees in this model.

The R-squared value of 0.541 suggests that the model explains a substantial proportion of the variation in PECEMPLOYEES.

In [10]:
model3 = smf.ols("ABSENT ~ SIZE + TURNOVER + LW + LONDON", data=df).fit()
print(model3.summary())

                            OLS Regression Results                            
Dep. Variable:                 ABSENT   R-squared:                       0.846
Model:                            OLS   Adj. R-squared:                  0.839
Method:                 Least Squares   F-statistic:                     130.3
Date:                Tue, 17 Mar 2026   Prob (F-statistic):           1.12e-37
Time:                        14:20:24   Log-Likelihood:                -137.80
No. Observations:                 100   AIC:                             285.6
Df Residuals:                      95   BIC:                             298.6
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      5.0004      0.232     21.573      0.0

In [11]:
model4 = smf.ols("PECEMPLOYEES ~ SIZE + TURNOVER + LW + LONDON", data=df).fit()
print(model4.summary())

                            OLS Regression Results                            
Dep. Variable:           PECEMPLOYEES   R-squared:                       0.755
Model:                            OLS   Adj. R-squared:                  0.744
Method:                 Least Squares   F-statistic:                     73.09
Date:                Tue, 17 Mar 2026   Prob (F-statistic):           3.74e-28
Time:                        14:20:39   Log-Likelihood:                -296.58
No. Observations:                 100   AIC:                             603.2
Df Residuals:                      95   BIC:                             616.2
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     55.1329      1.134     48.608      0.0

## Part (e) Effect of adding LONDON

After including the variable `LONDON` in both regression models, several changes can be observed.

In both models, `LONDON` is statistically significant and positively associated with the dependent variables. This suggests that firms located in London tend to have higher absenteeism and a higher proportion of early-career employees.

However, the coefficient for `LW` remains negative and statistically significant in both models, with only a slight change in magnitude.

This indicates that the effect of paying the Living Wage is robust and not driven by whether a firm is located in London.

Therefore, the inclusion of `LONDON` does not substantially alter the conclusions from earlier models. The evidence continues to support the Living Wage Foundation’s claims that firms paying the Living Wage have lower absenteeism and retain more experienced employees.

This suggests that there is little evidence of omitted variable bias from excluding `LONDON` in the earlier models.

In [12]:
df.corr()

,SIZE,TURNOVER,LONDON,LW,ECEMPLOYEES,ABSENT,PECEMPLOYEES
SIZE,1.000000,0.597439,-0.053606,0.179664,0.926994,-0.003814,-0.188300
TURNOVER,0.597439,1.000000,0.030486,0.097478,0.614845,0.079872,-0.023894
LONDON,-0.053606,0.030486,1.000000,0.003213,0.109288,0.827231,0.469426
LW,0.179664,0.097478,0.003213,1.000000,-0.056436,-0.381975,-0.725854
ECEMPLOYEES,0.926994,0.614845,0.109288,-0.056436,1.000000,0.218838,0.145098
ABSENT,-0.003814,0.079872,0.827231,-0.381975,0.218838,1.000000,0.673508
PECEMPLOYEES,-0.188300,-0.023894,0.469426,-0.725854,0.145098,0.673508,1.000000


## Part (f) Correlation and Multicollinearity

The correlation matrix shows that some variables are strongly correlated.

In particular, SIZE and ECEMPLOYEES have a very high positive correlation (approximately 0.93), indicating a strong linear relationship between these variables. TURNOVER is also moderately correlated with both SIZE and ECEMPLOYEES.

High correlation between independent variables can lead to multicollinearity, which is problematic because it makes it difficult to isolate the individual effect of each variable. It can also inflate standard errors and make coefficient estimates less reliable.

Although ECEMPLOYEES is not included as an independent variable in the regression models, there is still moderate correlation between SIZE and TURNOVER (around 0.60), which may introduce some degree of multicollinearity.

To address this issue, we could consider removing one of the correlated variables, combining them into a single measure, or using alternative techniques such as principal component analysis.

Overall, while multicollinearity may be present to some extent, it does not appear severe enough to invalidate the main conclusions of the analysis.

A chi-square test is appropriate in this case because both LONDON and LW are categorical (binary) variables, and we are testing for independence between them.

It is important to note that this analysis is based on observational data, and therefore we cannot infer causality. The results show associations rather than definitive causal relationships.

This may explain why some variables, such as SIZE and TURNOVER, are not statistically significant despite having theoretical relevance.